src/GIL/GIL.jl

```diff
@@ -9,6 +9,18 @@ module GIL
 
 using ..C: C
 
+# Ensure that only one Julia thread tries to acquire the Python GIL
+# PyGILState_Ensure and PyGILState_Release may not be thread safe.
```

Comment:
This reasoning is nonsense since those functions are explicitly thread safe. There may be good reasons for this PR, but this is not it.